prediction_market_agent/agents/known_outcome_agent/deploy.py

```diff
@@ -0,0 +1,82 @@
+import getpass
+from decimal import Decimal
+
+from prediction_market_agent_tooling.config import APIKeys
+from prediction_market_agent_tooling.deploy.agent import DeployableAgent
+from prediction_market_agent_tooling.deploy.constants import OWNER_KEY
+from prediction_market_agent_tooling.markets.agent_market import AgentMarket
+from prediction_market_agent_tooling.markets.data_models import BetAmount, Currency
+from prediction_market_agent_tooling.markets.markets import MarketType
+from prediction_market_agent_tooling.tools.utils import get_current_git_commit_sha
+from prediction_market_agent_tooling.tools.web3_utils import verify_address
+
+from prediction_market_agent.agents.known_outcome_agent.known_outcome_agent import (
+    Result,
+    get_known_outcome,
+)
+
+
+def market_is_saturated(market: AgentMarket) -> bool:
+    return market.p_yes > 0.95 or market.p_no > 0.95
+
+
+class DeployableKnownOutcomeAgent(DeployableAgent):
+    model = "gpt-3.5-turbo-0125"
+
+    def load(self) -> None:
+        self.markets_with_known_outcomes: dict[str, Result] = {}
+
+    def pick_markets(self, markets: list[AgentMarket]) -> list[AgentMarket]:
+        picked_markets: list[AgentMarket] = []
+        for market in markets:
+            # Assume very high probability markets are already known, and have
+            # been correctly bet on, and therefore the value of betting on them
+            # is low.
+            if not market_is_saturated(market=markets[0]):
+                answer = get_known_outcome(
+                    model=self.model,
+                    question=market.question,
+                    max_tries=3,
+                )
+                if answer.has_known_outcome():
+                    picked_markets.append(market)
+                    self.markets_with_known_outcomes[market.id] = answer.result
+
+        return picked_markets
+
+    def answer_binary_market(self, market: AgentMarket) -> bool:
+        # The answer has already been determined in `pick_markets` so we just
+        # return it here.
+        return self.markets_with_known_outcomes[market.id].to_boolean()
+
+    def calculate_bet_amount(self, answer: bool, market: AgentMarket) -> BetAmount:
+        if market.currency == Currency.xDai:
+            return BetAmount(amount=Decimal(0.1), currency=Currency.xDai)
```

Comment:
Should we make this configurable (e.g. ENV from Google Cloud) and thus editable after deployment?

Reply:
We currently have a way of changing agent source code after deployment via gcp, so I don't think that is needed. ATM we can update the commit hash that the agent's container installs (see pic), and click 'save and redeploy'
<img width="1113" alt="Screenshot 2024-03-20 at 15 18 58" src="https://github.com/gnosis/prediction-market-agent/assets/56087052/b578bf11-3931-46b0-a1be-960ecc2e76a9">